imageroot/systemd/user/nginx.service

```
@@ -0,0 +1,25 @@
+[Unit]
+Description=Podman  nginx.service
+BindsTo=webserver.service
+
+[Service]
+Environment=PODMAN_SYSTEMD_UNIT=%n
+EnvironmentFile=%S/state/environment
+WorkingDirectory=%S/state
+Restart=always
+TimeoutStopSec=70
+ExecStartPre=/bin/rm -f %t/nginx.pid %t/nginx.ctr-id
+ExecStartPre=%h/.config/bin/expand-vhosts
```

Runagent invocation should be put here (instead of inside the commands that are under .config/bin/: we assume the "agent environment" is already set for them by the caller)

```suggestion
ExecStartPre=runagent expand-vhosts
```

Also the path  `%h/.config` must be replaced with  `%S`· In other words for rootless modules `AGENT_INSTALL_DIR == %S` and `AGENT_STATE_DIR == %S/state`

---

done